### PR TITLE
Updated the app service related queries, added missing recommendations, updated index file for app services accordingly.

### PR DESCRIPTION
**Pull request by parthshiras**
…s, updated index file accordingly

# Overview/Summary

Updated the app service related queries, added missing recommendations, updated index file for app services accordingly.

## Related Issues/Work Items

[AB#29868](https://dev.azure.com/CSUSolEng/12ba7e46-a92a-47f6-ad6a-fba0479234a7/_workitems/edit/29868)
[AB#29870](https://dev.azure.com/CSUSolEng/12ba7e46-a92a-47f6-ad6a-fba0479234a7/_workitems/edit/29870)
[AB#29869](https://dev.azure.com/CSUSolEng/12ba7e46-a92a-47f6-ad6a-fba0479234a7/_workitems/edit/29869)
[AB#34345](https://dev.azure.com/CSUSolEng/12ba7e46-a92a-47f6-ad6a-fba0479234a7/_workitems/edit/34345)
[AB#34346](https://dev.azure.com/CSUSolEng/12ba7e46-a92a-47f6-ad6a-fba0479234a7/_workitems/edit/34346)

## This PR fixes/adds/changes/removes

1. Update index.md file for Web/sites to capture missing APRL queries
2. azure-resources/Web/sites/kql/493f6079-3bb6-4a56-96ba-ab3248474cb1.kql -> Cannot be validated with ARG
3. azure-resources/Web/sites/kql/9e6682ac-31bc-4635-9959-ab74b52454e6.kql -> Added to check if App service has atleast 2 prewarmed instances
4. azure-resources/Web/sites/kql/a7e8bb3d-8ceb-442d-b26f-007cd63f9ffc.kql -> Cannot be validated with ARG
5. azure-resources/Web/sites/kql/aab6b4a4-9981-43a4-8728-35c7ecbb746d.kql -> Added to check if no network restrictions are set
6. azure-resources/Web/sites/kql/fd049c28-ae6d-48f0-a641-cc3ba1a3fe1d.kql -> Added to check if Health check is enabled or not

### Breaking Changes

None

## As part of this pull request I have

- [x] Read the [Contribution Guide](https://azure.github.io/Azure-Proactive-Resiliency-Library-v2/contributing) and ensured this PR is compliant with the guide
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/issues) or ADO Work Items (Internal Only)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/tree/main)
- [ ] Ensured PR tests are passing
- [x] Performed testing and provided evidence (e.g. screenshot of output) for any changes associated to ARG queries
- [x] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

[Screenshots Appservice 05162024.zip](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/files/15338182/Screenshots.Appservice.05162024.zip)

### CODE REVIEW

File: azure-resources/Web/sites/kql/9e6682ac-31bc-4635-9959-ab74b52454e6.kql
Comment (ejhenry) [inline azure-resources/Web/sites/kql/9e6682ac-31bc-4635-9959-ab74b52454e6.kql:1]: Please ensure the query output matches the guidance in the contribution guide. Also, please include a screenshot in the PR of the successful testing of the query.

File: azure-resources/Web/sites/kql/aab6b4a4-9981-43a4-8728-35c7ecbb746d.kql
Comment (ejhenry) [inline azure-resources/Web/sites/kql/aab6b4a4-9981-43a4-8728-35c7ecbb746d.kql:1]: Please ensure the query output matches the guidance in the contribution guide. Also, please include a screenshot in the PR of the successful testing of the query.

File: azure-resources/Web/sites/kql/fd049c28-ae6d-48f0-a641-cc3ba1a3fe1d.kql
Comment (ejhenry) [inline azure-resources/Web/sites/kql/fd049c28-ae6d-48f0-a641-cc3ba1a3fe1d.kql:1]: Please ensure the query output matches the guidance in the contribution guide. Also, please include a screenshot in the PR of the successful testing of the query.

File: azure-resources/Web/sites/recommendations.yaml [hunk 1]
```diff
@@ -113,3 +113,63 @@
...
+
+- description: Health check
+  aprlGuid: fd049c28-ae6d-48f0-a641-cc3ba1a3fe1d
+  recommendationTypeId: null
+  recommendationControl: Other Best Practices
+  recommendationImpact: Medium
+  recommendationResourceType: Microsoft.Web/sites
+  recommendationMetadataState: Active
+  longDescription: |
+    Use Health Check for production workloads. Health check increases your application's availability by rerouting requests away from unhealthy instances, and replacing instances if they remain unhealthy. Your App Service plan should be scaled to two or more instances to fully utilize Health check. The Health check path should check critical components of your application. For example, if your application depends on a database and a messaging system, the Health check endpoint should connect to those components. If the application can't connect to a critical component, then the path should return a 500-level response code to indicate the app is unhealthy. Also, if the path does not return a response within 1 minute the health check ping is considered unhealthy.
```
Comment (ejhenry) [inline azure-resources/Web/sites/recommendations.yaml:125]: longDescription has a limit of 300 characters, including spaces.

File: azure-resources/Web/sites/recommendations.yaml [hunk 2]
```diff
@@ -113,3 +113,63 @@
...
+  potentialBenefits: Enhanced reliability via automation
+  pgVerified: false
+  publishedToLearn: false
+  publishedToAdvisor: false
+  automationAvailable: arg
+  tags: null
+  learnMoreLink:
```
Comment (ejhenry) [inline azure-resources/Web/sites/recommendations.yaml:132]: Please include only a single learn more link

File: azure-resources/Web/sites/recommendations.yaml [hunk 3]
```diff
@@ -113,3 +113,63 @@
   learnMoreLink:
     - name: Configure web apps in Azure App Service
       url: "https://learn.microsoft.com/azure/app-service-web/web-sites-configure"
+
+- description: Health check
```
Comment (ejhenry) [inline azure-resources/Web/sites/recommendations.yaml:117]: This should be worded in the form of a recommendation

File: azure-resources/Web/sites/recommendations.yaml [hunk 4]
```diff
@@ -113,3 +113,63 @@
...
+  automationAvailable: arg
+  tags: null
+  learnMoreLink:
+    - name: Monitor the health of App Service instances
+      url: "https://learn.microsoft.com/en-us/azure/app-service/monitor-instances-health-check?tabs=dotnet#enable-health-check"
+    - name: The Ultimate Guide to Running Healthy Apps in the Cloud
+      url: "https://azure.github.io/AppService/2020/05/15/Robust-Apps-for-the-cloud.html"
+
+
+- description: No network access restrictions defined
```
Comment (ejhenry) [inline azure-resources/Web/sites/recommendations.yaml:139]: This should be worded in the form of a recommendation

**Review by ejhenry — COMMENTED** [review]
Thanks for the contribution @parthshiras. Just a few items that need to be addressed in the PR.

File: azure-resources/Web/sites/kql/9e6682ac-31bc-4635-9959-ab74b52454e6.kql
Comment (parthshiras) [reply azure-resources/Web/sites/kql/9e6682ac-31bc-4635-9959-ab74b52454e6.kql:1]: Addressed. 

Added tags and removed kind properties.

Comment (parthshiras) [reply azure-resources/Web/sites/kql/9e6682ac-31bc-4635-9959-ab74b52454e6.kql:1]: @eric - this PR included multiple queries, hence all the screenshots are added in a zip folder. But sharing the updated query output with this comment.

image

File: azure-resources/Web/sites/recommendations.yaml (hunk 3 above)
Comment (parthshiras) [reply azure-resources/Web/sites/recommendations.yaml:117]: Reworded to Enable Health check for App Services.

File: azure-resources/Web/sites/recommendations.yaml (hunk 1 above)
Comment (parthshiras) [reply azure-resources/Web/sites/recommendations.yaml:125]: Updated to be within 300 characters.

File: azure-resources/Web/sites/recommendations.yaml (hunk 2 above)
Comment (parthshiras) [reply azure-resources/Web/sites/recommendations.yaml:132]: Kept the most relevant link.

File: azure-resources/Web/sites/recommendations.yaml (hunk 4 above)
Comment (parthshiras) [reply azure-resources/Web/sites/recommendations.yaml:139]: Updated the description.

File: azure-resources/Web/sites/kql/aab6b4a4-9981-43a4-8728-35c7ecbb746d.kql
Comment (parthshiras) [reply azure-resources/Web/sites/kql/aab6b4a4-9981-43a4-8728-35c7ecbb746d.kql:1]: Removed kind and added tags per requirement. Screenshot below -

Query -
image

Relevant network restrictions on the app which are not enabled -
image

File: azure-resources/Web/sites/kql/fd049c28-ae6d-48f0-a641-cc3ba1a3fe1d.kql
Comment (parthshiras) [reply azure-resources/Web/sites/kql/fd049c28-ae6d-48f0-a641-cc3ba1a3fe1d.kql:1]: Removed kind and added tags in the property.

Screenshot of the query -
image
image



Relevant app service and screenshot that shows health check is not enabled-
image
image

**Comment by parthshiras** [comment]
> @parthshiras do you have questions about the feedback or require assistance with the changes?



> @parthshiras do you have questions about the feedback or require assistance with the changes?

@ejhenry - updated queries/files per comments. thank you for the review.

File: azure-resources/Web/sites/recommendations.yaml [hunk 5]
```diff
@@ -113,3 +113,60 @@
...
+  pgVerified: false
+  publishedToLearn: false
+  publishedToAdvisor: false
+  automationAvailable: arg
+  tags: null
+  learnMoreLink:
+    - name: Monitor the health of App Service instances
+      url: "https://learn.microsoft.com/en-us/azure/app-service/monitor-instances-health-check?tabs=dotnet#enable-health-check"
+
+- description: Check if network access restrictions are defined
```
Comment (ejhenry) [inline azure-resources/Web/sites/recommendations.yaml:136]:
```suggestion
- description: Configure network access restrictions
```

File: azure-resources/Web/sites/recommendations.yaml [hunk 6]
```diff
@@ -113,3 +113,60 @@
...
+
+- description: Enable Health check for App Services
+  aprlGuid: fd049c28-ae6d-48f0-a641-cc3ba1a3fe1d
+  recommendationTypeId: null
+  recommendationControl: Other Best Practices
+  recommendationImpact: Medium
+  recommendationResourceType: Microsoft.Web/sites
+  recommendationMetadataState: Active
+  longDescription: |
+    Use Health Check for production workloads. Health check increases your application's availability by rerouting requests away from unhealthy instances, and replacing instances if they remain unhealthy. The Health check path should check critical components of your application.
```
Comment (ejhenry) [inline azure-resources/Web/sites/recommendations.yaml:125]:
```suggestion
    Use Health Check for production workloads. Health check increases your application's availability by rerouting requests away from unhealthy instances, and replacing instances if they remain unhealthy. The Health check path should check critical components of your application.
```

**Review by ejhenry — APPROVED** [review]
LGTM 👍

**Pull request merged**